pandas/core/groupby/groupby.py

```diff
@@ -4740,7 +4740,7 @@ def _wrap_transformed_output(self, output, names=None):
 
     def _wrap_agged_blocks(self, items, blocks):
         if not self.as_index:
-            index = np.arange(blocks[0].values.shape[1])
+            index = np.arange(blocks[0].values.shape[-1])
```

Comment:
@reidy-p pushed a simplification. but maybe need some additional tests that do this when a column is selected

e.g. ``df.groupby('id', as_index=False)['foo'].first()``

Reply:
Nice simplification. I added some new tests.